routes/topics.js

```diff
@@ -16,6 +16,8 @@ router.get('/follow', isAuth, topicsController.getFollowTopic);
 router.get('/latest', validate(TopicValidation.latestPost), isAuth, topicsController.getLatestPost);
 router.put('/follow', isAuth, topicsController.putFollowTopic);
 router.put('/follow-v2', isAuth, topicsController.followTopicV2);
+router.post('/check-name', validate(TopicValidation.checkName), isAuth, topicsController.checkName);
+router.post('/create', validate(TopicValidation.create), isAuth, topicsController.create);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new routes `/check-name` and `/create` are both POST methods. However, the `/check-name` route seems to be used for checking if a topic name is available, which doesn't change any server state and is more suitable for a GET method. Consider changing this to a GET method for better RESTful API design.
```diff
- router.post('/check-name', validate(TopicValidation.checkName), isAuth, topicsController.checkName);
+ router.get('/check-name', validate(TopicValidation.checkName), isAuth, topicsController.checkName);
```
Also, ensure that the `isAuth` middleware is properly validating the user's authentication status before proceeding to the controller functions. This is crucial for security reasons.


<!-- This is an auto-generated comment by OSS CodeRabbit -->